File: test_kpts.jl
#!/usr/bin/env julia
# Test k-point ordering between Python and Julia

nk = 5

# Julia approach
kx = range(-0.5, 0.5, length=nk)
ky = range(-0.5, 0.5, length=nk)
kgrid_julia = [[x, y] for x in kx, y in ky]
kgrid_flat_julia = vec(kgrid_julia)

println("Julia k-points (first 10):")
for i in 1:10
    println("  $i: ", kgrid_flat_julia[i])
end

File: test_kpts.py
#!/usr/bin/env python3
# Test k-point ordering

import numpy as np

nk = 5

# Python approach
kx = np.linspace(-0.5, 0.5, nk)
ky = np.linspace(-0.5, 0.5, nk)
kgrid = np.meshgrid(kx, ky, indexing='ij')
kgrid_flat = np.stack(kgrid, axis=-1).reshape(-1, 2)

print("Python k-points (first 10):")
for i in range(10):
    print(f"  {i+1}: {kgrid_flat[i]}")
